File: CinemaNet.Abstractions/Models/Movie/MovieDetails.cs
namespace CinemaNet.Abstractions.Models.Movie;

public class MovieDetails
{
    public string Id { get; init; }

    public string Name { get; init; }

    public int Duration { get; init; }

    public DateTime ReleaseDate { get; init; }

    public string Category { get; init; }

    public string Description { get; init; }

    public string[] MediaUrls { get; init; }

    public string Director { get; init; }
    
    public string[] Staff { get; init; }
}

File: CinemaNet.Web/src/app/modules/movie/models/movie-details.ts
export class MovieDetails {
    public id!: string;
    public name!: string;
    public duration!: number;
    public releaseDate!: Date;
    public category!: string;
    public description!: string;
    public mediaUrls!: string[];
    public director!: string;
    public staff!: string[];

    public static createEmpty(): MovieDetails {
        return {
            id: '',
            name: '',
            duration: 0,
            releaseDate: new Date(1, 1, 1),
            category: '',
            description: '',
            mediaUrls: [],
            director: '',
            staff: []
          }
    }
}
